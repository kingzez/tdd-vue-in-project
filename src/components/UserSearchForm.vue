<template>
  <div>
    <form @submit.prevent="$emit('submitted', username)">
      <el-input type="text" v-model="username" placeholder="search for user" class="search-form__input">
        <el-button native-type="submit" slot="append" icon="el-icon-search" class="search-form__button">提交</el-button>
      </el-input>
    </form>
  </div>
</template>
<script>
export default {
  name: 'UserSearchForm',
  data() {
    return {
      username: ''
    }
  }
}
</script>