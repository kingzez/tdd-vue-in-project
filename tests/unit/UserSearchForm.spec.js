import { shallowMount, createLocalVue, mount } from '@vue/test-utils'
import ElementUI from 'element-ui'
import UserSearchForm from '@/components/UserSearchForm'

const localVue = createLocalVue()
localVue.use(ElementUI)

describe('UserSearchForm', () => {
  const build = () => {
    const options = { localVue }
    const wrapper = shallowMount(UserSearchForm, options)
    const wrapperMounted = mount(UserSearchForm, options)

    return {
      wrapper,
      wrapperMounted,
      input: () => wrapper.find('.search-form__input'),
      inputMounted: () => wrapperMounted.find('input'),
      button: () => wrapperMounted.find('.search-form__button')
    }
  }

  test('render the component', () => {
    // arrange
    const { wrapper } = build()

    // assert
    expect(wrapper.html()).toMatchSnapshot()
  })


  test('render main child components', () => {
    const { input, button } = build()

    expect(input().exists()).toBe(true)
    expect(button().exists()).toBe(true)
  })

  test('call "submitted" event when submitting form', () => {
    // arrange
    const expectedUser = 'kingzez'
    const { wrapperMounted, inputMounted, button } = build()
    inputMounted().element.value = expectedUser

    // act
    inputMounted().trigger('input')
    button().trigger('click')
    button().trigger('submit')

    // assert
    expect(wrapperMounted.emitted().submitted[0]).toEqual([expectedUser])
  })
})
