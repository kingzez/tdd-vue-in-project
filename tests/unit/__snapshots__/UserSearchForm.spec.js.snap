// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSearchForm render the component 1`] = `
<div>
  <form>
    <el-input-stub value="" type="text" autocomplete="off" validateevent="true" placeholder="search for user" class="search-form__input">
      <el-button-stub type="default" icon="el-icon-search" nativetype="submit" class="search-form__button">提交</el-button-stub>
    </el-input-stub>
  </form>
</div>
`;
